huntsman/camera/pyro.py

```diff
@@ -177,73 +185,51 @@ def take_exposure(self,
+                value 'Dark Frame' instead of 'Light Frame'. Set dark to None to disable the
+                `IMAGETYP` keyword entirely.
             blocking (bool, optional): If False (default) returns immediately after starting
                 the exposure, if True will block until it completes.
 
         Returns:
-            threading.Event: Event that will be set when exposure is complete
+            threading.Event: Event that will be set when exposure is complete.
 
         """
-        assert self.is_connected, self.logger.error("Camera must be connected for take_exposure!")
-
-        assert filename is not None, self.logger.warning("Must pass filename for take_exposure")
-
-        if not self._exposure_event.is_set():
-            msg = "Attempt to take exposure on {} while one already in progress.".format(self)
-            raise error.PanError(msg)
-
-        # Clear event now to prevent any other exposures starting before this one is finished.
-        self._exposure_event.clear()
-
-        # Make sure proxy is in async mode
-        Pyro4.asyncproxy(self._proxy, asynchronous=True)
-
         # Start the exposure
-        self.logger.debug(f'Taking {seconds} second exposure on {self.name}: {filename}')
+        self.logger.debug(f'Taking {seconds} second exposure on {self}: {filename}')
 
         # Remote method call to start the exposure
-        exposure_result = self._proxy.take_exposure(seconds=seconds,
-                                                    filename=filename,
-                                                    dark=bool(dark),
-                                                    *args,
-                                                    **kwargs)
-
-        # Start a thread that will set an event once exposure has completed
-        exposure_thread = Timer(interval=get_quantity_value(seconds, u.s) + self.readout_time,
-                                function=self._async_wait,
-                                args=(exposure_result,
-                                      'exposure',
-                                      self._exposure_event,
-                                      self._timeout))
-        exposure_thread.start()
+        self._proxy.take_exposure(seconds=seconds,
+                                  filename=filename,
+                                  dark=bool(dark),
+                                  *args,
+                                  **kwargs)
 
+        max_wait = get_quantity_value(seconds, u.second) + self.readout_time + self._timeout
         if blocking:
-            self._exposure_event.wait()
+            success = self._exposure_event.wait(timeout=max_wait)
+            if not success:
+                self._timeout_response("exposure")
+        else:
+            # If the remote exposure fails after starting in such a way that the event doesn't
+            # doesn't get set then calling code could wait forever. Have a local timeout thread
+            # to be safe.
```

Comment:
Removing extra "doesn't"

```suggestion
            # If the remote exposure fails after starting in such a way that the event doesn't
            # get set then calling code could wait forever. Have a local timeout thread to be safe.
```